templates/module.build.ps1

```diff
@@ -103,13 +103,22 @@ task Publish {
 task PublishNuget {
     $newVersion = New-Object version -ArgumentList 1, 0, 0, $BuildNumber
     "Version is $newVersion"
-    $Public  = @(Get-ChildItem -Path $ModulePath\Public\*.ps1 -ErrorAction SilentlyContinue)
-    $Functions = $public.basename
-    Update-ModuleManifest -Path $ModulePath\$ModuleName.psd1 -ModuleVersion $newVersion -FunctionsToExport $Functions
+
+    $PublicScriptPath = "$ModulePath\Public"
+    $psFilter = "*.ps1"
+    $functions = ( Get-ChildItem -Path $PublicScriptPath -Filter $psFilter -Recurse -ErrorAction SilentlyContinue | Select-String -Pattern "function" )
```

Comment:
I came across a situation where there was a comment with the word functions in it. If you change the select string to `Select-String -Pattern "function "` that will avoid picking up those lines, though if there is a comment with just function we will still have a false positive.